src/Sulu/Component/HttpKernel/SuluKernel.php

```diff
@@ -34,39 +49,144 @@
+                (isset($envs['all']) || isset($envs[$this->environment]))
+                // and if not registered for other context.
+                && !isset($envs[$this->reversedContext])
+            ) {
+                yield new $class();
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader)
+    {
+        $container->addResource(new FileResource($this->getProjectDir() . '/config/bundles.php'));
+        $confDir = $this->getProjectDir() . '/config';
+
+        $this->load($loader, $confDir . '/{packages}/*');
+        $this->load($loader, $confDir . '/{packages}/' . $this->environment . '/**/*');
+        $this->load($loader, $confDir . '/{services}');
+        $this->load($loader, $confDir . '/{services}_' . $this->environment);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureRoutes(RouteCollectionBuilder $routes)
+    {
+        $confDir = $this->getProjectDir() . '/config';
+
+        $this->import($routes, $confDir . '/{routes}/*');
+        $this->import($routes, $confDir . '/{routes}/' . $this->environment . '/**/*');
+        $this->import($routes, $confDir . '/{routes}');
+    }
+
+    protected function load(LoaderInterface $loader, $glob)
+    {
+        $configExtensions = $this->getConfigExtensions();
+        $notReversedContext = $this->getNotReversedContext();
+        $configFiles = glob($glob . $configExtensions, GLOB_BRACE);
+        $excludedConfigFiles = glob($glob . $notReversedContext, GLOB_BRACE);
+
+        foreach ($configFiles as $resource) {
+            if (!in_array($resource, $excludedConfigFiles)) {
+                $loader->load($resource);
+            }
+        }
+    }
+
+    protected function import(RouteCollectionBuilder $routes, $glob)
+    {
+        $configExtensions = $this->getConfigExtensions();
+        $notReversedContext = $this->getNotReversedContext();
+
+        $configFiles = glob($glob . $configExtensions, GLOB_BRACE);
+        $excludedConfigFiles = glob($glob . $notReversedContext, GLOB_BRACE);
+
+        foreach ($configFiles as $resource) {
+            if (!in_array($resource, $excludedConfigFiles)) {
+                $routes->import($resource, '/');
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCacheDir()
+    {
+        return $this->getProjectDir() . DIRECTORY_SEPARATOR
+            . 'var' . DIRECTORY_SEPARATOR
+            . 'cache' . DIRECTORY_SEPARATOR
+            . $this->context . DIRECTORY_SEPARATOR
+            . $this->environment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLogDir()
+    {
+        return $this->getProjectDir() . DIRECTORY_SEPARATOR
+            . 'var' . DIRECTORY_SEPARATOR
+            . 'logs' . DIRECTORY_SEPARATOR
+            . $this->context;
+    }
+
+    private function getNotReversedContext()
```

Comment:
`getNotReversedContextSuffix`? It doesn't really return the context here, that confused me a bit. Or do you even have a better idea?

Reply:
the not is also confusing. I would say `getReversedConfigExtensions()`